platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/ProfileMigratable.java

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.migratable;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.apache.karaf.features.FeaturesService;
+import org.boon.json.JsonFactory;
+import org.boon.json.ObjectMapper;
+import org.codice.ddf.admin.application.service.ApplicationService;
+import org.codice.ddf.migration.ExportMigrationContext;
+import org.codice.ddf.migration.ImportMigrationContext;
+import org.codice.ddf.migration.Migratable;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Class used to migrate the state of applications, features, and bundles. */
+public class ProfileMigratable implements Migratable {
```

Comment:
I used that name to align this functionality with the profile install command which not only provides the ability to install predefined profiles but also custom made one. This migratable does restore the same profile that was used before (and any other changes made along the way) so I think it still applies.

Reply:
This is actually something I was thinking about while reviewing. I agree with @tbatie. Profile is technically correct but it is an overloaded word in our codebase. The Profile Install command is different because it uses the word "install" which is more specific. I think maybe `InstallationProfileMigratable` would be better. (This is the same mess we got into with Migration export versus Catalog export).

Reply:
I don't think I'd call the state the system exports a profile since it can have random bundles installed & features turned on/off. All existing profiles, including any custom ones integrators would make in production, are determined at build time. This seems a lot more indeterministic.

Reply:
You might be forgetting about the profile "list" command. This is the profile migratable element which is used to "migrate" the profile. I think it is in the same lines as the others.

I do agree that the word profile might be overloaded in many places but so far we have used to reference a set of "things" to be installed. This is in these lines. Starting to call it something else without taking the time to change the other too to align would be, in my opinion, more confusing. As for adding the word "installation", I am not sure about that since we are migrating the profile and not going through the installation. I know it is semantics but we are making a difference to the customer between having to do the installation and performing a migration import.

Reply:
The profile list command is not a good example because that is undocumented and should be probably be renamed to include "install" or "installation" in the name. I still don't agree that we should leave this as is.

We wouldn't be adding the word "installation" because that's what the user is doing. We'd be adding the word "installation" because the user is importing an "installation profile".

Reply:
@tbatie So does the profile:install command, it can install a random set of bundles and/or features. For that matter, even the pre-defined profiles could be defined as such.

@brjeter Not disagreeing that it is not a good example but the fact is that until we revamped the whole thing, we chose to identify a set of features, bundles, and/or apps as a profile and not an "installation profile" so short of changing all of them, I would rather keep it consistent ;-).

Reply:
@paouelle the inconsistency is already there. We might as well start working towards the direction we want to go in now instead of waiting. See the documentation where we refer to these as `installation profiles`: http://www.codice.org/ddf/documentation.html#_setup_types

Reply:
I do prefer `InstallationProfileMigratable` over `ProfileMigratable`.

Reply:
I will leave that up to a different PR to rename the concept to something else than profile. For consistency, we have:
- install profile
- list profile
- migration/migrate profile (since profile is now also an option on the migrate command in an upcoming PR)

Reply:
Could you create a tech debt ticket for it, if you haven't already?